packages/plan-form-core/src/formHelpers/helpers.ts

```diff
@@ -637,17 +624,15 @@ export function generatePlanDefinition(
     ...defaultEnvConfig,
     ...envConfigs,
   };
-  const planIdentifier =
-    formValue.identifier && formValue.identifier !== '' // is this an existing plan?
-      ? formValue.identifier
-      : generateNameSpacedUUID(moment().toString(), configs.planUuidNamespace);
-
-  const planVersion =
-    formValue.identifier && formValue.identifier !== '' // is this an existing plan?
-      ? isNaN(parseInt(formValue.version, 10)) // is the existing version valid?
-        ? parseInt(configs.defaultPlanVersion, 10) + 1
-        : parseInt(formValue.version, 10) + 1
-      : formValue.version;
+  const planIdentifier = formValue?.identifier // is this an existing plan?
+    ? formValue.identifier
+    : generateNameSpacedUUID(moment().toString(), configs.planUuidNamespace);
+
+  const planVersion = formValue?.identifier // is this an existing plan?
+    ? isNaN(parseInt(formValue.version, 10)) // is the existing version valid?
```

Comment:
Will this `isNaN(parseInt(formValue.version, 10)` fail when `formValue.identifier -== ''`?

Reply:
The ternary operator checks for truthiness so `formValue?.identifier ?` evaluates to false for both `formValue.identifier === undefined` and `formValue.identifier === ''`. `formValue.identifier === ''` evaluates to falsy so `isNaN(parseInt(formValue.version, 10)` will be skipped